### PR TITLE
DOC: stats.ttest_1samp: update example 

### LINKED ISSUE #15696: DOC: False information in docs - scipy.stats.ttest_1samp
### Describe your issue.

The documentation page for ttest_1samp provides example code and the results from running that code. However, the results are orders of magnitude off from what I get running the same code, and the results are also wrongly interpreted, saying the high p-value means that the null hypothesis should be rejected.

### Reproducing Code Example

```python
import numpy as np
from scipy import stats
rng = np.random.default_rng()
rvs = stats.norm.rvs(loc = 5, scale = 10, size = (50, 2), random_state = rng)
stats.ttest_1samp(rvs, 5.)
stats.ttest_1samp(rvs, 0.)
```

### Error message

```shell
Here's the result I got for the last two lines

Ttest_1sampResult(statistic=array([ 1.07810207, -0.38435094]), pvalue=array([0.28626709, 0.70238137]))
Ttest_1sampResult(statistic=array([4.38743017, 2.96982928]), pvalue=array([6.08529768e-05, 4.60362353e-03]))

And here's what's on the current doc page

Ttest_1sampResult(statistic=array([-2.09794637, -1.75977004]), pvalue=array([0.04108952, 0.08468867]))
Ttest_1sampResult(statistic=array([1.64495065, 1.62095307]), pvalue=array([0.10638103, 0.11144602]))
```

### SciPy/NumPy/Python version information

1.7.0 1.21.0 sys.version_info(major=3, minor=9, micro=5, releaselevel='final', serial=0)

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-15696

#### What does this implement/fix?
After documentation examples switched to `np.random.default_rng` in gh-13863, the random data no longer cooperated with the story we intended to tell about `stats.ttest_1samp`. I took this opportunity to rewrite the examples with a more explicit criterion for rejecting the null hypothesis and an example using a one-sided alternative.

### CODE REVIEW

**Comment by mdhaber** [comment]
@latteprog Thanks for reporting gh-15696. Does this address it?

**Comment by mdhaber** [comment]
For convenience, [here](https://44165-1460385-gh.circle-artifacts.com/0/html-scipyorg/reference/generated/scipy.stats.ttest_1samp.html#scipy.stats.ttest_1samp) is the rendered version.

image
image

**Review by tupui — APPROVED** [review]
Thanks @mdhaber. It reads good and I like the addition at the end. CI failure is unrelated...

What we are actually missing in general for all test is to clearly write H0, H1, test applicability and explain what it all means for everyone to understand. When you are new (or not, I've seen senior researchers mixing up everything too) to this, it's not trivial.

**Comment by mdhaber** [comment]
Thanks @tupui. Yup, that's why I'm rewriting these rather than just changing the numbers. I guess I'll open a tracking issue.

**Review by rkern — COMMENTED** [review]
I started this review before it got an approval. I might as well finish it, but feel free to ignore the recommendations; I also approve.

In general, I don't think I'd like to see the full working out of the logic of p-value hypothesis tests in the examples of these functions. It's both too much (for this function) and too little (to do justice to the subtleties of hypothesis testing). I certainly don't want to see this kind of detail in _each_ of our p-value test functions, and now this one is the odd function out. I would really like to see that kind of material moved out to the more topic-oriented portion of our docs to talk about our p-value hypothesis testing infrastructure in general and introducing the statistical background that's needed to reference (e.g. to interpret `alternative`). `ttest_1samp` is a good function to use in such a tutorial because it's one of the simplest, so I think this text is a good start. Then I'd keep each of these docstrings focused on more concise examples, possibly demonstrating the things that are unique to each test rather than repeating generalities.

File: scipy/stats/_stats_py.py
Comment (mdhaber) [reply scipy/stats/_stats_py.py:5950]: My understanding is that sample means, not the underlying data, needs to follow a normal distribution. For sufficiently large samples, sample means from a uniform distribution do follow a normal distribution due to the CLT.

**Comment by mdhaber** [comment]
> In general, I don't think I'd like to see the full working out of the logic of p-value hypothesis tests in the examples of these functions.

Would you draft an ideal (in your opinion) example? Perhaps we can compromise, but (perhaps obviously) I thought this was the right balance.

**Comment by rkern** [comment]
Honestly, I'd just have adjusted `size=(50, 2)` to `size=(100, 2)` (and/or reduce the `scale`) to make the original narrative work out and called it a day.

I don't think this can be the right balance for this docstring, because then it would be the right balance for the docstrings of all of our NHST functions, and that's very repetitive. I think the general topic deserves the space of a tutorial or other "User's Guide" style rather than "API Reference" genre. Then all of the docstrings can just refer to that general topic documentation. There's a lot of subtlety to the topic, and taking the time to explain everything with nice visualizations for all of the `alternative` options is better than half-explaining it multiple times across many docstrings. But I've been officially Not A Fan of NHST for decades and thus have no real energy to volunteer myself for that, so take that for what it's worth.

File: scipy/stats/_stats_py.py
Comment (rkern) [reply scipy/stats/_stats_py.py:5950]: Well, it increasingly _approximates_ a normal distribution, while the means of samples drawn from normals obey it exactly at all sample sizes. But yeah, once you're above a dozen or so samples, the test statistic from uniform samples obeys a T distribution fairly well (other distributions can be more problematic).

However, most treatments I see talk about the assumption being about the distribution of the sample itself and recommend using normality tests on the sample to determine if it's appropriate to apply a T-test. To use a uniform distribution in the example flies in the face of that advice (which is reasonable advice despite it not being an absolute rule) and deserves a mention, if only to not confuse novices who've been taught that advice. But that mention is mostly a distraction for the introductory example that can be avoided.

**Comment by mdhaber** [comment]
OK. Maybe we'll revert.
@latteprog, what do you think?

**Comment by tupui** [comment]
IMO we should not revert since there is an improvement over the previous state. I do agree, and wrote above about it, that we could have a central place to discuss hypothesis testing in general. Once this is done we can simplify these and link to this page.

**Comment by rkern** [comment]
Agreed. No real reason to revert, just a suggestion for where to go next.

**Comment by mdhaber** [comment]
OK. I think I can get on board with that. I've planned on writing a tutorial about `nan_policy`, `axis`, etc. for all the stats functions to go along with gh-14651, so I could move this sort of thing over there when I get to it.

**Comment by josef-pkt** [comment]
> However, most treatments I see talk about the assumption being about the distribution of the sample itself and recommend using normality tests on the sample to determine if it's appropriate to apply a T-test. To use a uniform distribution in the example flies in the face of that advice (which is reasonable advice despite it not being an absolute rule) and deserves a mention, if only to not confuse novices who've been taught that advice. But that mention is mostly a distraction for the introductory example that can be avoided.

That's a good reason NOT to use the normal distribution in the example.
Talking about the underlying distribution as requirement for sample mean testing is misleading. T-test is the best alternative in most general cases, when we are not sure what the underlying distribution is.

**Pull request merged**
